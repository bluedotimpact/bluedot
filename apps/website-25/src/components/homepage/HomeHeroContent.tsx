import {
  HeroH1,
  HeroH2,
} from '@bluedot/ui';
import Rive, { Fit, Layout } from '@rive-app/react-canvas';
import clsx from 'clsx';

const HomeHeroContent: React.FC<{ className?: string }> = ({ className }) => (
  <div className={clsx('home-hero-content pb-5 bg-bluedot-darker', className)}>
    <div className="home-hero-content__animation-container w-[350px] max-w-[80vw] aspect-4/3 mx-auto">
      <Rive
        src="/animations/herobluedot_hero.riv"
        layout={new Layout({ fit: Fit.Cover })}
      />
    </div>
    <HeroH1 className="home-hero-content__heading slide-up-fade-in">
      The expertise you need to shape safe AI
    </HeroH1>
    <HeroH2 className="home-hero-content__subheading slide-up-fade-in" style={{ animationDelay: '100ms' }}>
      Learn alongside thousands of professionals through comprehensive courses designed with leading AI safety experts.
    </HeroH2>
  </div>
);

export default HomeHeroContent;
