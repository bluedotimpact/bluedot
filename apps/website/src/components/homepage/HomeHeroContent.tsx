import clsx from 'clsx';
import { H1 } from '@bluedot/ui/src/Text';
import HomeGraduateSection from './HomeGraduateSection';

const HomeHeroContent: React.FC<{ className?: string }> = ({ className }) => (
  <>
    {/* Hero with integrated nav and alumni */}
    <div className={clsx('relative w-full', className)}>
      {/* Main Hero Container */}
      <div
        className="relative flex flex-col min-h-[653px] min-[680px]:max-[1023px]:min-h-[659px] min-[1024px]:max-[1279px]:min-h-[738px] min-[1280px]:max-[1439px]:min-h-[821px] min-[1440px]:min-h-[821px] w-full -scale-x-100"
        style={{
          /* Tailwind can't handle multiple backgrounds with blend modes */
          background: 'linear-gradient(180deg, rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, 0.2) 100%), url(\'/images/agi-strategy/hero-banner-split.png\'), #FFFFFF',
          backgroundSize: 'cover',
          backgroundPosition: 'center',
          backgroundBlendMode: 'color-burn, normal, normal',
        }}
      >
        {/* Noise Overlay */}
        <div className="absolute inset-0 z-10 opacity-50 mix-blend-overlay pointer-events-none bg-[url('/images/homepage/noise.svg')] bg-repeat" />

        {/* Nav spacer */}
        <div className="relative z-20 min-h-[60px] min-[680px]:max-[1023px]:min-h-[60px] min-[1024px]:min-h-[76px] -scale-x-100" />

        {/* Content Container */}
        <div className="relative z-20 flex-1 flex flex-col justify-end px-5 py-12 min-[680px]:max-[1023px]:justify-center min-[680px]:max-[1023px]:px-8 min-[680px]:max-[1023px]:py-20 min-[1024px]:max-[1279px]:justify-center min-[1024px]:max-[1279px]:px-12 min-[1024px]:max-[1279px]:pt-32 min-[1024px]:max-[1279px]:pb-24 min-[1280px]:max-[1439px]:justify-start min-[1280px]:max-[1439px]:pt-40 min-[1280px]:max-[1439px]:pb-[120px] min-[1280px]:max-[1439px]:px-12 min-[1440px]:justify-center min-[1440px]:px-6 min-[1440px]:py-0 -scale-x-100">
          <div className="w-full max-w-[1488px] mx-auto min-[1440px]:px-6">
            {/* Text container with responsive dimensions */}
            <div className="w-full max-w-[280px] min-[680px]:max-[1023px]:max-w-[616px] min-[680px]:max-[1023px]:min-h-[348px] min-[1024px]:max-[1279px]:w-[768px] min-[1024px]:max-[1279px]:max-w-screen-md min-[1024px]:max-[1279px]:min-h-[347px] min-[1280px]:max-[1439px]:w-[768px] min-[1280px]:max-[1439px]:max-w-screen-md min-[1280px]:max-[1439px]:min-h-[374px] min-[1440px]:w-[900px] min-[1440px]:max-w-[900px] min-[1440px]:min-h-[374px] mx-auto flex flex-col justify-center items-center gap-8 text-center text-white">
              <H1
                className="w-full text-[40px] min-[680px]:max-[1023px]:text-[56px] min-[1024px]:max-[1279px]:w-[682px] min-[1024px]:max-[1279px]:text-[64px] min-[1280px]:max-[1439px]:w-[682px] min-[1280px]:max-[1439px]:text-[72px] min-[1440px]:w-[682px] min-[1440px]:text-[72px] leading-[115%] font-normal slide-up-fade-in flex items-center tracking-[-1px] text-white"
              >
                We help you have a positive impact on the trajectory of AI
              </H1>
              <p
                className="text-size-sm min-[680px]:max-[1023px]:text-[20px] min-[1024px]:text-[20px] leading-[155%] font-normal w-full max-w-[280px] min-[680px]:max-[1023px]:max-w-[616px] min-[1024px]:max-[1279px]:max-w-screen-md min-[1280px]:max-[1439px]:max-w-screen-md min-[1440px]:max-w-[900px] mx-auto tracking-[-0.005em]"
                style={{
                  /* Tailwind doesn't support OpenType font-feature-settings for stylistic alternates */
                  fontFeatureSettings: '"ss02" on',
                }}
              >
                BlueDot is the leading talent accelerator for beneficial AI and societal resilience. We run courses, help people land jobs, organise events all over the world, and accelerate entrepreneurs to start new companies
              </p>
            </div>
          </div>
        </div>

        {/* Alumni Section - at bottom of hero */}
        <div className="relative z-20 -scale-x-100">
          <HomeGraduateSection />
        </div>
      </div>
    </div>
  </>
);

export default HomeHeroContent;
