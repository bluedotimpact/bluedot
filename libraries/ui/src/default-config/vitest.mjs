/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
