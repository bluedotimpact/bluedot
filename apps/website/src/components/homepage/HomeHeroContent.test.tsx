import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import HomeHeroContent from './HomeHeroContent';

describe('HomeHeroContent', () => {
  test('renders as expected', () => {
    const { container } = render(<HomeHeroContent />);
    expect(container).toMatchSnapshot();
  });
});
