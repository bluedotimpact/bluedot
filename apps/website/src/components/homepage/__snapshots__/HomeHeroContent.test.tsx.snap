// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeHeroContent > renders as expected 1`] = `
<div>
  <div
    class="relative w-full"
  >
    <div
      class="relative flex flex-col min-h-[653px] min-[680px]:max-[1023px]:min-h-[659px] min-[1024px]:max-[1279px]:min-h-[738px] min-[1280px]:max-[1439px]:min-h-[821px] min-[1440px]:min-h-[821px] w-full -scale-x-100"
      style="background-size: cover; background-position: center center; background-blend-mode: color-burn, normal, normal;"
    >
      <div
        class="absolute inset-0 z-10 opacity-50 mix-blend-overlay pointer-events-none bg-[url('/images/homepage/noise.svg')] bg-repeat"
      />
      <div
        class="relative z-20 min-h-[60px] min-[680px]:max-[1023px]:min-h-[60px] min-[1024px]:min-h-[76px] -scale-x-100"
      />
      <div
        class="relative z-20 flex-1 flex flex-col justify-end px-5 py-12 min-[680px]:max-[1023px]:justify-center min-[680px]:max-[1023px]:px-8 min-[680px]:max-[1023px]:py-20 min-[1024px]:max-[1279px]:justify-center min-[1024px]:max-[1279px]:px-12 min-[1024px]:max-[1279px]:pt-32 min-[1024px]:max-[1279px]:pb-24 min-[1280px]:max-[1439px]:justify-start min-[1280px]:max-[1439px]:pt-40 min-[1280px]:max-[1439px]:pb-[120px] min-[1280px]:max-[1439px]:px-12 min-[1440px]:justify-center min-[1440px]:px-6 min-[1440px]:py-0 -scale-x-100"
      >
        <div
          class="w-full max-w-[1488px] mx-auto min-[1440px]:px-6"
        >
          <div
            class="w-full max-w-[280px] min-[680px]:max-[1023px]:max-w-[616px] min-[680px]:max-[1023px]:min-h-[348px] min-[1024px]:max-[1279px]:w-[768px] min-[1024px]:max-[1279px]:max-w-screen-md min-[1024px]:max-[1279px]:min-h-[347px] min-[1280px]:max-[1439px]:w-[768px] min-[1280px]:max-[1439px]:max-w-screen-md min-[1280px]:max-[1439px]:min-h-[374px] min-[1440px]:w-[900px] min-[1440px]:max-w-[900px] min-[1440px]:min-h-[374px] mx-auto flex flex-col justify-center items-center gap-8 text-center text-white"
          >
            <h1
              class="bluedot-h1 not-prose w-full text-[40px] min-[680px]:max-[1023px]:text-[56px] min-[1024px]:max-[1279px]:w-[682px] min-[1024px]:max-[1279px]:text-[64px] min-[1280px]:max-[1439px]:w-[682px] min-[1280px]:max-[1439px]:text-[72px] min-[1440px]:w-[682px] min-[1440px]:text-[72px] leading-[115%] font-normal slide-up-fade-in flex items-center tracking-[-1px] text-white"
            >
              We help you have a positive impact on the trajectory of AI
            </h1>
            <p
              class="text-size-sm min-[680px]:max-[1023px]:text-[20px] min-[1024px]:text-[20px] leading-[155%] font-normal w-full max-w-[280px] min-[680px]:max-[1023px]:max-w-[616px] min-[1024px]:max-[1279px]:max-w-screen-md min-[1280px]:max-[1439px]:max-w-screen-md min-[1440px]:max-w-[900px] mx-auto tracking-[-0.005em]"
              style="font-feature-settings: "ss02" on;"
            >
              BlueDot is the leading talent accelerator for beneficial AI and societal resilience. We run courses, help people land jobs, organise events all over the world, and accelerate entrepreneurs to start new companies
            </p>
          </div>
        </div>
      </div>
      <div
        class="relative z-20 -scale-x-100"
      >
        <section
          class="w-full h-[106px] min-[680px]:h-[91px] bg-white border-b-hairline border-color-divider flex items-center"
        >
          <div
            class="w-full flex items-center justify-center 2xl:justify-end p-5 sm:px-6 md:px-12 2xl:pl-[272px] 2xl:pr-[48px]"
          >
            <div
              class="flex flex-col sm:flex-row items-center gap-4 sm:gap-5 w-full max-w-[1200px] 2xl:max-w-none"
            >
              <p
                class="bluedot-p not-prose text-[#13132E] text-[16px] leading-[26px] whitespace-nowrap flex-shrink-0"
              >
                Our 5,000+ alumni work at
              </p>
              <div
                class="w-full sm:flex-1 inline-flex flex-nowrap overflow-hidden [mask-image:_linear-gradient(90deg,transparent_0,_black_12.63%,_black_80.26%,transparent_100%)]"
              >
                <ul
                  class="flex items-center justify-start [&_li]:mx-8 [&_img]:max-w-none animate-infinite-scroll list-none"
                >
                  <li>
                    <img
                      alt="OpenAI"
                      class="h-6"
                      src="/images/third-party-logos/openai.svg"
                    />
                  </li>
                  <li>
                    <img
                      alt="Anthropic"
                      class="h-6 !h-3"
                      src="/images/third-party-logos/anthropic.svg"
                    />
                  </li>
                  <li>
                    <img
                      alt="Google DeepMind"
                      class="h-6"
                      src="/images/third-party-logos/deepmind.svg"
                    />
                  </li>
                  <li>
                    <img
                      alt="AI Security Institute"
                      class="h-6"
                      src="/images/third-party-logos/aisi.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="United Nations"
                      class="h-6"
                      src="/images/third-party-logos/un.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Amnesty International"
                      class="h-6"
                      src="/images/third-party-logos/amnesty.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Time"
                      class="h-6"
                      src="/images/third-party-logos/time.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="NATO"
                      class="h-6"
                      src="/images/third-party-logos/nato.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="OECD"
                      class="h-6"
                      src="/images/third-party-logos/oecd.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Stanford HAI"
                      class="h-6"
                      src="/images/third-party-logos/hai.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Apple"
                      class="h-6"
                      src="/images/third-party-logos/apple.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Harvard Kennedy School"
                      class="h-6"
                      src="/images/third-party-logos/harvard.png"
                    />
                  </li>
                </ul>
                <ul
                  aria-hidden="true"
                  class="flex items-center justify-start [&_li]:mx-8 [&_img]:max-w-none animate-infinite-scroll list-none"
                >
                  <li>
                    <img
                      alt="OpenAI"
                      class="h-6"
                      src="/images/third-party-logos/openai.svg"
                    />
                  </li>
                  <li>
                    <img
                      alt="Anthropic"
                      class="h-6 !h-3"
                      src="/images/third-party-logos/anthropic.svg"
                    />
                  </li>
                  <li>
                    <img
                      alt="Google DeepMind"
                      class="h-6"
                      src="/images/third-party-logos/deepmind.svg"
                    />
                  </li>
                  <li>
                    <img
                      alt="AI Security Institute"
                      class="h-6"
                      src="/images/third-party-logos/aisi.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="United Nations"
                      class="h-6"
                      src="/images/third-party-logos/un.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Amnesty International"
                      class="h-6"
                      src="/images/third-party-logos/amnesty.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Time"
                      class="h-6"
                      src="/images/third-party-logos/time.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="NATO"
                      class="h-6"
                      src="/images/third-party-logos/nato.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="OECD"
                      class="h-6"
                      src="/images/third-party-logos/oecd.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Stanford HAI"
                      class="h-6"
                      src="/images/third-party-logos/hai.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Apple"
                      class="h-6"
                      src="/images/third-party-logos/apple.png"
                    />
                  </li>
                  <li>
                    <img
                      alt="Harvard Kennedy School"
                      class="h-6"
                      src="/images/third-party-logos/harvard.png"
                    />
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
